### PR TITLE
Release 3.10.24 - Shellcheck refactoring, syntax changes and regex bug fix

### PR DESCRIPTION
**Pull request by KGjidodaj**
Used ShellCheck to scan the scripts, fixed-replaced old legacy code and standardized string length check using the -z operator.

### Added
- 

### Changed (non-breaking)
- Split exports into two parts to avoid error masking..
- Resolved ShellCheck warnings like logic fixes that could result in errors.

### Changed (BREAKING)
-

### Deprecated
-

### Removed
-

### Fixed
Fixed regex bug by removing literal quotes from =~ pattern matching (.*executionRoleArn.*), allowing (FARGATE_JQ_FILTER) to be built correctly.

### Security
- Run local testing with no apparent errors.

---

### Feature PR Checklist
- [x] Documentation: Checked, nothing to be updated.
- [x] Unit tests created or updated: Kept existing test.bats file mostly the same, updated ecs-deploy script according to shellcheck standards helping with edge cases or potential errors and fixed regex bug.

### Release PR Checklist
- [x] Update version number in `ecs-deploy` script: From 3.10.23 - 3.10.24
- [x] Give the PR a title of `Release _._._ - Summary of change` (using whatever
      the new version number is, plus a succinct summary of what changed)

### CODE REVIEW

**Comment by briskt** [comment]
@KGjidodaj thank you for the contribution. The copilot review flagged some problems that need to be addressed. Tests probably don't reveal the problem because there are no tests that actually invoke the `aws` command using the `AWS_ECS` variable.

**Comment by KGjidodaj** [comment]
Hi @briskt, thanks for the review! I checked the Copilot findings and ShellCheck quoting did break the multi-word execution for $AWS_ECS. I pushed a new commit that reverts quoting where it is not needed and actually causes errors. I also updated the PR description. Just let me know if any other error occurs and if anything else is needed.

File: ecs-deploy [hunk 1]
```diff
@@ -129,9 +129,13 @@ function assumeRole() {
                     --role-arn "${AWS_ASSUME_ROLE}" \
                     --role-session-name "$(date +"%s")")
 
-   export AWS_ACCESS_KEY_ID=$(echo $temp_role | jq .Credentials.AccessKeyId | xargs)
-   export AWS_SECRET_ACCESS_KEY=$(echo $temp_role | jq .Credentials.SecretAccessKey | xargs)
-   export AWS_SESSION_TOKEN=$(echo $temp_role | jq .Credentials.SessionToken | xargs)
+AWS_ACCESS_KEY_ID=$(echo "$temp_role" | jq .Credentials.AccessKeyId | xargs)
+export AWS_ACCESS_KEY_ID
```
Comment (briskt) [inline ecs-deploy]: Please correct the indentation. These lines are in the body of `assumeRole` so they should be indented one level.

File: ecs-deploy [hunk 2]
```diff
@@ -213,39 +217,39 @@ function parseImageName() {
     # - tagOrDigest
     # - emptyOrSha
     # If a group is missing it will be an empty string
-    if [[ "x$TAGONLY" == "x" ]]; then
+    if [[ "$TAGONLY" == "" ]]; then
```
Comment (briskt) [inline ecs-deploy]: This could even be further simplified
```suggestion
    if [[ -z "$TAGONLY" ]]; then
```

File: ecs-deploy [hunk 3]
```diff
@@ -401,10 +405,10 @@ function createNewTaskDefJson() {
 
     # Updated jq filters for AWS Fargate
     REQUIRES_COMPATIBILITIES=$(echo "${DEF}" | jq -r '. | select(.requiresCompatibilities != null) | .requiresCompatibilities[]')
-    if `echo ${REQUIRES_COMPATIBILITIES[@]} | grep -q "FARGATE"`; then
+    if echo "$REQUIRES_COMPATIBILITIES" | grep -q "FARGATE"; then
       FARGATE_JQ_FILTER='requiresCompatibilities: .requiresCompatibilities'
 
-      if [[ ! "$NEW_DEF_JQ_FILTER" =~ ".*executionRoleArn.*" ]]; then
+      if [[ ! "$NEW_DEF_JQ_FILTER" =~ .*executionRoleArn.* ]]; then
```
Comment (briskt) [inline ecs-deploy:411]: This looks like a bugfix. It's probably worth mentioning it in the PR description.

**Review by briskt — CHANGES_REQUESTED** [review]

**Comment by KGjidodaj** [comment]
I believe any errors has been fixed and every syntax changes has been applied. If there is anything else I have forgotten by all means tell me.

**Review by briskt — APPROVED** [review]
Thanks!
